### ISSUE TITLE
ndtv.com - Anti adb

### ISSUE THREAD

**Issue by gotitbro** [uBlockOrigin/uAssets#188]
Anti adb warning is back on the website `http://www.ndtv.com/`, the **uBlock filters** fix is not working anymore.

**Error URL:** http://www.ndtv.com/delhi-news/paytms-vijay-shekhar-sharma-not-alone-in-seeking-to-escape-delhi-smog-1624096?pfrom=home-lateststories

_Using Chrome 54 with latest Ublock filters._

**Comment by gotitbro** [comment]
@gorhill Can you please help with this, can't seem to figure out what is causing this.

**Comment by IDKwhattoputhere** [comment]
Found this quick fix through [the logger](https://github.com/gorhill/uBlock/wiki/The-logger):
`@@||s3.amazonaws.com/khachack/ima3.js$script,domain=ndtv.com`
Try it and report back.

**Comment by gotitbro** [comment]
Yes, it works, don't think the `$script` & `domain=ndtv.com` are needed nothing else there on the page.

**Comment by IDKwhattoputhere** [comment]
Filter options like that prevent accidentally whitelisting more requests than needed and prevent the filter from being abused ([example](https://github.com/uBlockOrigin/uAssets/issues/39#issuecomment-258941003)).

**Comment by gotitbro** [comment]
@IDKwhattoputhere I see, thanks for the help.

**Comment by gorhill** [comment]
That solution should be reported to EasyList.

**Comment by gotitbro** [comment]
@okiehsch @gorhill I am hiding ad spaces on http://profit.ndtv.com through this filter: 
`ndtv.com##div[class^=ad300]`

While the above filter does not cause any issues, it does lead to an anti adb warning on the stocks section of the website such as here: http://profit.ndtv.com/stock/interglobe-aviation-ltd_indigo

Any idea on how I can still use the filter to hide ad spaces and stop this anti adb warning. Thank You.

**Comment by okiehsch** [comment]
`ndtv.com##script:inject(abort-on-property-write.js, canRunAdsN)`

**Comment by gotitbro** [comment]
@okiehsch Thanks, the filter works. Similar warnings are also encountered on http://www.ndtv.com & https://sports.ndtv.com. But these seem to be using `canRunAds` and not `canRunAdsN`.

Error URL: 
http://www.ndtv.com/convergence/ndtv/advertise/ndtv_leaderboard.aspx
https://sports.ndtv.com/icc-champions-trophy-2017/pakistan-stun-india-in-final-to-claim-maiden-champions-trophy-crown-1713891

What should be done here?

**Comment by okiehsch** [comment]
With uBO default settings and filters there is no issue on my end, it only occurs if I add your custom
filter `ndtv.com##div[class^=ad300]`, so I don't think any filter should be added to `uBO-filters`.
If you add the custom filter you will have to add
````
ndtv.com##script:inject(abort-on-property-write.js, canRunAdsN)
ndtv.com##script:inject(abort-on-property-write.js, canRunAds)
````
to your own filter list.

**Comment by gotitbro** [comment]
@okiehsch Sure, no problem. I was having this problem on my side as described in the initial post.

Another question, is there syntax for aborting script that begins with something. For example as in the above case for aborting a script that begins with `canRunAds`?

**Comment by okiehsch** [comment]
No if you use for example `abort-on-property-write.js`  you have to use the specific argument.

**Comment by okiehsch** [comment]
And just to be clear, I think using `ndtv.com##div[class^=ad300]` to hide empty ad spaces could be 
added to a potential `uBO-annoyance` list alongside the then nedded `script:inject` filters.

**Comment by gotitbro** [comment]
@okiehsch I don't think these are annoyances they are simply blank ad spaces. These can be added to **_uBlock filters_** as it already contains such filters to hide blank ad spaces.

For example: https://github.com/uBlockOrigin/uAssets/blob/master/filters/filters.txt#L713

**Comment by okiehsch** [comment]
Well I would classify a blank ad space as an annoyance :wink:, but if @gorhill thinks otherwise,
let him decide.

**Comment by SMed79** [comment]
If you start hunting blank spaces this will just have the effect to increase the list over time.

**Comment by gotitbro** [comment]
@SMed79 @okiehsch Not hunting but if multiple ad spaces/elements can be hidden by a few/single filter such as here it can be added. though not a necessity.

**Comment by gotitbro** [comment]
Sponsored ads in the right sidebar on http://gadgets.ndtv.com.

**Error URL:** http://gadgets.ndtv.com/telecom/news/reliance-jio-slams-airtel-vodafones-baseless-allegations-on-trais-iuc-ruling-1753009

Can be blocked by `||hotdeals360.com^$third-party`.

**Comment by SMed79** [comment]
> Can be blocked by ||hotdeals360.com^$third-party

It's not an adserver add only
`||hotdeals360.com/static/js/kpwidgetweb.js`

**Comment by gotitbro** [comment]
It is a coupon website and is its being used on a third-party website its only for ads, at least that is how these coupon sites work.

I would limit the filter like this though: `||hotdeals360.com^$script,domain=ndtv.com`

**Comment by SMed79** [comment]
> They also have a few ads on their homepage

`gadgets.ndtv.com##.container > .topstorypanel`
or 
`gadgets.ndtv.com##.topstorypanel:has:(.heading_shp_wdgt)`

**Comment by gotitbro** [comment]
@gorhill uBO causes an issue on **_ndtv.com_** whereby I am unable close a section that is displayed by clicking the close button. The close button works without any issues when uBO is disabled.

**Screencapture:** `https://streamable.com/v7vvw`

**Error URL:** 
`https://www.ndtv.com/gurgaon-news/gurgaon-schoolboy-murder-student-detained-by-cbi-say-sources-1772591`

I am unable to figure out what is causing this. Please have a look at this.

_**Note:** Using uBO 1.14.16 on Chrome 62 with the default filters and settings._

**Comment by okiehsch** [comment]
`@@||ads.ndtv1.com/*/jquery_dfp_infinite.js$script,domain=ndtv.com`

**Comment by gotitbro** [comment]
@okiehsch That works. Should be added to filters.

**Comment by ghajini** [comment]
@ryanbr @mapx-
Annoyance which covers almost page partially hiding page contents
Firefox android  65,  ublock origin 1.18.6

`https://khabar.ndtv.com/news/lok-sabha-elections-2019/setback-for-bjp-in-northeast-25-leaders-including-ministers-and-mla-quit-party-in-a-week-over-ticket-2010270?pfrom=home-topstories`

Filter

`khabar.ndtv.com##.break_fbmsg > [src^="data:image/svg+xml;utf8;base64,"]`

**Comment by mapx-** [comment]
it seems blocking taboola breaks all the page content, test these filters see if there are ads, other breakages
```
khabar.ndtv.com##.break_fbmsg
@@||taboola.com^$domain=khabar.ndtv.com
```

**Comment by ghajini** [comment]
I don't know taboola is used for displaying legit content or not
Don't whitelist taboola completely thatway,  it just invites sponsored ads

**Comment by mapx-** [comment]
on your side using only this filter is all fine ?
`khabar.ndtv.com##.break_fbmsg`

**Comment by ghajini** [comment]
khabar.ndtv.com##.break_fbmsg
fixes 
https://github.com/uBlockOrigin/uAssets/issues/188#issuecomment-474726865

**Comment by mapx-** [comment]
so, you need taboola as well. The problem is I get no such sponsored stuff. Could be geo based.

**Comment by ghajini** [comment]
Try Indian ip

**Comment by jspenguin2017** [comment]
I can't reproduce. Any specific test link?

**Comment by ghajini** [comment]
Try Firefox android exclusively for this....unable to reproduce with user agent switcher on desktop.....

**Comment by ghajini** [comment]
ok i will open new issue in ordert o have better understanding of issue here

**Comment by jspenguin2017** [comment]
I can't test on that setup.
